src/disco/components/Addon/index.js

```diff
@@ -121,31 +124,48 @@ export class AddonBase extends React.Component {
   }
 
   getThemeImage() {
-    const { getBrowserThemeData, i18n, name, previewURL } = this.props;
-    if (this.props.type === ADDON_TYPE_THEME) {
+    const { type } = this.props.addon;
+
+    if (ADDON_TYPE_THEMES.includes(type)) {
+      const { addon, getBrowserThemeData, i18n } = this.props;
+      const { name, previewURL, previews } = addon;
+
+      let imageUrl = getPreviewImage(previews);
+
+      if (!imageUrl && type === ADDON_TYPE_THEME) {
+        imageUrl = previewURL;
+      }
+
+      const headerImage = (
+        <img
+          alt={sprintf(i18n.gettext('Preview of %(name)s'), { name })}
+          className="Addon-theme-header-image"
+          src={imageUrl}
+        />
+      );
+
       /* eslint-disable jsx-a11y/href-no-hash, jsx-a11y/anchor-is-valid */
-      return (
+      return type === ADDON_TYPE_THEME ? (
         <a
-          href="#"
           className="theme-image"
           data-browsertheme={getBrowserThemeData()}
+          href="#"
           onClick={this.installTheme}
         >
-          <img
-            src={previewURL}
-            alt={sprintf(i18n.gettext('Preview of %(name)s'), { name })}
-          />
+          {headerImage}
         </a>
+      ) : (
+        <React.Fragment>{headerImage}</React.Fragment>
```

Comment:
Do you really need a fragment here?

Reply:
If I don't have this, it breaks. This is kind of temporary tho anyway, b/c in the functionality PR, we should add back the `a` wrapper tag which has an onClick (a special case for static themes). I realize your work might override that with your work but I am just going off what I know my changes would be for now :)

Reply:
mmh, what exactly is breaking? A fragment renders nothing, and I believe `headerImage` is a valid element.

Reply:
I see this in console:

Caught application error: Invariant Violation: "Objects are not valid as a React child (found: object with keys {headerImage}). If you meant to render a collection of children, use an array instead.

![Alt text](https://monosnap.com/image/3bx6ETf0AAyQXWvJsnchB9YcHhZeel.png)

Reply:
mmh, this works for me:

```js
      return type === ADDON_TYPE_THEME ? (
        <a
          className="theme-image"
          data-browsertheme={getBrowserThemeData()}
          href="#"
          onClick={this.installTheme}
        >
          {headerImage}
        </a>
      ) : (
        headerImage
      );

```